File: src/main/java/org/styl/gravitus/ui/Screen.java
package org.styl.gravitus.ui;

import java.awt.Color;
import java.awt.Component;
import java.awt.Graphics;
import java.awt.Graphics2D;
import java.awt.Point;

import javax.swing.ImageIcon;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.SwingConstants;

import org.styl.gravitus.Specs;

import lombok.Getter;
import lombok.Setter;

@SuppressWarnings("serial")
@Getter @Setter
public class Screen extends JPanel {

	private JLabel fps;
	private JLabel zoomOut;
	private JLabel zoomIn;
	
	public Screen() {
		super();
	}

	public void init() {
		
		fps = new JLabel();
		fps.setForeground(Color.WHITE);
		fps.setHorizontalTextPosition(SwingConstants.RIGHT);
		fps.setBounds((int) (Specs.instance.frameX - 60), 8, 70, 10);
		add(fps);
		
		zoomIn = new JLabel("r");
		zoomIn.setBounds((int) (Specs.instance.frameX - 60), 30, 16, 16);
		zoomIn.setIcon(new ImageIcon("resources/icons/plus.png"));
		add(zoomIn);
		
		zoomOut = new JLabel("a");
		zoomOut.setBounds((int) (Specs.instance.frameX - 35), 30, 16, 16);
		zoomOut.setIcon(new ImageIcon("resources/icons/minus.png"));
		add(zoomOut);

		
		setBackground(Color.BLACK);
		setLayout(null);

	}

	@Override
	protected void paintComponent(Graphics g) {
		super.paintComponent(g);

		if(Specs.instance.orbitTrails) {
			g.setColor(Color.WHITE);
			Graphics2D g2d = (Graphics2D) g;
			
			for(Component cmpnt : this.getComponents()) {
				if(cmpnt instanceof SpaceObjectUIWrapper) {
					SpaceObjectUIWrapper wrap = (SpaceObjectUIWrapper) cmpnt;

					Point prev = null;
					for(Point p : wrap.getPastPositions()) {
						if(prev != null) {
							g2d.drawLine(
								(int)(p.getX()),
								(int)(p.getY()),
								(int)(prev.getX()),
								(int)(prev.getY())
							);
						}
						prev = p;
					}
					
				}
			}
		}
		
	}

}

File: src/main/java/org/styl/gravitus/ui/Screen.kt
package org.styl.gravitus.ui

import org.styl.gravitus.Specs
import java.awt.*
import java.awt.event.ActionListener
import javax.swing.*

@Suppress("serial")
class Screen : JPanel() {

	var listener: ActionListener? = null

	lateinit var fps: JLabel
		private set
	lateinit var zoomOut: JButton
		private set
	lateinit var zoomIn: JButton
		private set

	fun init() {
		fps = JLabel().apply {
			foreground = Color.WHITE
			horizontalTextPosition = SwingConstants.RIGHT
			setBounds((Specs.instance?.frameX ?: 800) - 60, 8, 70, 10)
		}
		add(fps)

		zoomIn = JButton().apply {
			setContentAreaFilled(false)
			setBorderPainted(false)
			cursor = Cursor(Cursor.HAND_CURSOR)
			setBounds((Specs.instance?.frameX ?: 800) - 60, 30, 16, 16)
			icon = ImageIcon("resources/icons/plus.png")
			actionCommand = "zoomIn"
			listener?.let { addActionListener(it) }
		}
		add(zoomIn)

		zoomOut = JButton().apply {
			setContentAreaFilled(false)
			setBorderPainted(false)
			cursor = Cursor(Cursor.HAND_CURSOR)
			setBounds((Specs.instance?.frameX ?: 800) - 35, 30, 16, 16)
			icon = ImageIcon("resources/icons/minus.png")
			actionCommand = "zoomOut"
			listener?.let { addActionListener(it) }
		}
		add(zoomOut)

		background = Color.BLACK
		layout = null
	}

	override fun paintComponent(g: Graphics) {
		super.paintComponent(g)

		if (Specs.instance?.orbitTrails == true) {
			g.color = Color.WHITE
			val g2d = g as Graphics2D

			for (cmpnt in components) {
				if (cmpnt is SpaceObjectUIWrapper) {
					var prev: Point? = null
					for (p in cmpnt.pastPositions) {
						prev?.let { g2d.drawLine(p.x, p.y, it.x, it.y) }
						prev = p
					}
				}
			}
		}
	}
}
